### PR TITLE
chore: add warnings for deprecated BrowserWindow methods

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Prepare for the removal of unnecessary forwarders.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes
Notes: none

### CODE REVIEW

**Review by nornagon — CHANGES_REQUESTED** [review]
I don't think removing these APIs is worth the trouble.

**Comment by miniak** [comment]
@nornagon does it make sense to keep the refactoring, which forwards all the listed methods in the for loop instead of writing the code manually?

**Comment by miniak** [comment]
@nornagon also what about only deprecating the methods, which are not documented?

**Comment by nornagon** [comment]
I'm fine with removing undocumented methods. We can do that without warning, generally speaking. As for auto-forwarding vs. listing the method names in an array, I think it's fine to keep them as straightforward functions without an array / automatic forwarding once the number is much smaller (4-5 or so).

File: docs/breaking-changes.md [hunk 1]
```diff
@@ -14,6 +14,24 @@ This document uses the following convention to categorize breaking changes:
...
 
+### Deprecated: `BrowserWindow` APIs
+
+The following APIs have been deprecated:
+
+* `browserWindow.loadURL`
+* `browserWindow.loadFile`
+* `browserWindow.reload`
+* `browserWindow.showDefinitionForSelection`
+* `browserWindow.capturePage`
```
Comment (nornagon) [inline docs/breaking-changes.md:25]: Actually having thought about it, I'm ok with these three being deprecated and removed. loadURL/loadFile should stay though.

**Review by codebytere — COMMENTED** [review]
Was there some form of discussion about removing these methods in a future version? While i'm ok with removing the undocumented ones, these were largely here for convenience, which they provided. I don't think that having this type of duplication is inherently a negative unless there's an active negative consequence to their being there, which is not the case here that I am aware.

**Pull request closed without merging**